tests/std/tests/P1208R6_source_location/test.cpp

```diff
@@ -0,0 +1,130 @@
+    constexpr auto x = source_location::current();
+    assert(x.line() == 36);
+    assert(x.column() == 41);
+    assert(x.function_name() == "local_test"sv);
+    assert(string_view{x.file_name()}.ends_with(R"(tests\std\tests\P1208R6_source_location\test.cpp)"sv));
+}
+
+constexpr void global_test() {
+    assert(g.line() == 10);
+    assert(g.column() == 37);
+    assert(g.function_name() == ""sv);
+    assert(string_view{g.file_name()}.ends_with(R"(tests\std\tests\P1208R6_source_location\test.cpp)"sv));
+}
+
+constexpr void argument_test(unsigned line, unsigned column, source_location x = source_location::current()) {
+    assert(x.line() == line);
+    assert(x.column() == column);
+    assert(x.function_name() == "test"sv);
+    assert(string_view{x.file_name()}.ends_with(R"(tests\std\tests\P1208R6_source_location\test.cpp)"sv));
+}
+
+constexpr void sloc_constructor_test() {
+    s x;
+    assert(x.loc.line() == 58);
+    assert(x.loc.column() == 7);
+    assert(x.loc.function_name() == "sloc_constructor_test"sv);
+    assert(string_view{x.loc.file_name()}.ends_with(R"(tests\std\tests\P1208R6_source_location\test.cpp)"sv));
+}
+
+constexpr void different_constructor_test() {
+    s x{1};
+    assert(x.loc.line() == 14);
+    assert(x.loc.column() == 5);
+    assert(x.loc.function_name() == "s"sv);
+    assert(string_view{x.loc.file_name()}.ends_with(R"(tests\std\tests\P1208R6_source_location\test.cpp)"sv));
+}
+
+constexpr void sub_member_test() {
+    s2 s;
+    assert(s.x.loc.line() == 19);
+    assert(s.x.loc.column() == 7);
+    assert(s.x.loc.function_name() == "s2"sv);
+    assert(string_view{s.x.loc.file_name()}.ends_with(R"(tests\std\tests\P1208R6_source_location\test.cpp)"sv));
+}
+
+constexpr void lambda_test() {
+    auto l = [loc = source_location::current()]() { return loc; };
+    auto x = l();
+    assert(x.line() == 82);
+    assert(x.column() == 21);
+    assert(x.function_name() == "lambda_test"sv);
+    assert(string_view{x.file_name()}.ends_with(R"(tests\std\tests\P1208R6_source_location\test.cpp)"sv));
+}
+
+template <class T>
+constexpr source_location function_template() {
+    return source_location::current();
+}
+
+constexpr void function_template_test() {
+    auto x1 = function_template<void>();
+    assert(x1.line() == 92);
+    assert(x1.column() == 29);
+    assert(x1.function_name() == "function_template"sv);
+    assert(string_view{x1.file_name()}.ends_with(R"(tests\std\tests\P1208R6_source_location\test.cpp)"sv));
+
+    auto x2 = function_template<int>();
+    assert(x1.line() == x2.line());
+    assert(x1.column() == x2.column());
+    assert(string_view{x1.function_name()} == string_view{x2.function_name()});
+    assert(string_view{x1.file_name()} == string_view{x2.file_name()});
+}
+
+constexpr bool test() {
+    copy_test();
+    // tab_test();
+    global_test();
+    local_test();
+    argument_test(114, 5);
+    auto loc = std::source_location::current();
+    argument_test(115, 38, loc);

```

Comment:
These hard coded line values may get invalid pretty quickly while editing this file. I would suggest doing something like that, though I leave it up to you.
```suggestion
    argument_test(__LINE__, 5);
    auto loc = std::source_location::current();
    argument_test(__LINE__ - 1, 38, loc);
```